Realm/RLMRealm.mm

```diff
@@ -484,11 +371,11 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration
     realm->_dynamic = dynamic;
 
     // protects the realm cache and accessors cache
-    static std::mutex& initLock = *new std::mutex();
-    std::lock_guard<std::mutex> lock(initLock);
+    static auto& initLock = *new RLMUnfairMutex;
```

Comment:
Why do we want this to be static? I.e., isn't this locking in the case of multiple realms?

Reply:
The scope of this lock can probably be reduced, but all of this code has always been written under the assumption that it's called with a global lock held and changing that is an unrelated project.